app/services/users/delete.rb

```diff
@@ -0,0 +1,36 @@
+module Users
+  class Delete
+    def initialize(user)
+      @user = user
+    end
+
+    def call
+      delete_comments
+      delete_articles
+      delete_user_activity
+      user.unsubscribe_from_newsletters
+      CacheBuster.new.bust("/#{user.username}")
+      user.delete
```

Comment:
what happens if any of these fail? I was thinking of a transaction in case one fails and the job has to be retried but maybe it's not necessary

Reply:
Yeah I agree, something here to account for possible edge cases to an extent greater than we might in some other scenarios makes sense.

Reply:
I suppose if any of these fails we can continue the process later when the issue is resolved. I don't think we need to re-create (via rollback) the content that had already been deleted before the error.
But I think that we need to be notified in case a deletion job fails, not just rely on the DJ web interface. What would be a best way to notify about it?

Reply:
We currently use Slack notifications for stuff that needs actions, but I think we should run this through logging/observability and then define further alerts from there. cc: @mstruve

Reply:
100% yes we should be sending this kind of event to Honeycomb so that we can aggregate on it and then alert when we hit Thresholds!!! What I need to do is create a wrapper for the lib Honeycomb Client so that we can easily send data without having to worry about the actual implementation. Since I do not have that set up yet can you log a record of the failure for now via Rails so that it gets sent to Timber? Please! Once I create a Honeycomb client wrapper I will be looking at all our Rails logging to move it over and thats when I will get it sent to Honeycomb.

Reply:
I will add logging, but I'm worried that it won't be noticed there.

Reply:
If we want it noticed we can easily add an alert to Timber for it and I can set that up if you want. Honestly, I am less concerned about it alerting us immediately and more concerned that if a problem arises we have the logs we need to be able to go back and debug it.

Reply:
Okay let's add the log and then add the timber alert to make sure we get a slack message if there is a problem with this. Seems like the optimal solution. Ensures we have logs, ensures we get the notice.